src/Relations/BelongsToMany.php

```diff
@@ -116,7 +116,7 @@ public function sync($ids, $detaching = true)
         ];
 
         if ($ids instanceof Collection) {
-            $ids = $ids->modelKeys();
+            $ids = $this->parseIds($ids);
```

Comment:
As mentioned in #2666, `$ids->modelKeys()` returns `_id` field or primary key. However, `$this->parseIds($ids)` returns `relatedKey` which is the primary key by default.